pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -34,24 +35,23 @@ def __init__(self, amp_level: str = "O2") -> None:
         super().__init__()
         self.backend = AMPType.APEX
         self.amp_level = amp_level
+        self._connected = False
 
     def master_params(self, optimizer: Optimizer) -> _PARAMETERS:
         return amp.master_params(optimizer)
 
-    def connect(
-        self,
-        model: Module,
-        optimizers: List[Optimizer],
-        lr_schedulers: List[Any],
-    ) -> Tuple[Module, List[Optimizer], List[Any]]:
-        """Connects the precision plugin to the training process,
-        configures apex and reinits the schedulers
-        """
-        if model.device.type != "cuda":
-            return model, optimizers, lr_schedulers
-        model, optimizers = self.configure_apex(amp, model, list(optimizers), self.amp_level)
-        self.reinit_scheduler_properties(optimizers, lr_schedulers)
-        return model, optimizers, lr_schedulers
+    def pre_dispatch(self) -> None:
+        return super().pre_dispatch()
+
+    def dispatch(self, trainer: "pl.Trainer") -> None:
+        if not self._connected:
```

Comment:
can you quickly remind us why dispatch is called multiple times ( so you have the connected flag)?

Reply:
yes, I think this should only be called once.

Reply:
Yes, exactly.

Reply:
ok so I understand now. it is called once within a fit, so that's good. but we don't teardown and .test() etc. later on we can't configure again. 

got it.